### PR TITLE
fix: Dynamic code now accepts bare Unity Object calls

### PR DESCRIPTION
**Pull request by hatayama**
## Summary
- Dynamic code snippets can now use bare `Object.Instantiate` and related Unity object calls without qualifying `UnityEngine.Object`.
- Existing custom `Object` aliases are respected, so snippets that define their own alias do not get duplicate using directives.

## User Impact
- Before this change, snippets could fail with an ambiguous `Object` reference because the wrapper imported both `System` and `UnityEngine`.
- After this change, common Unity-style snippets using `Object` compile as expected in script mode.

## Changes
- Add a default `Object = UnityEngine.Object` alias to wrapped script snippets.
- Skip alias injection when a snippet already declares an `Object` alias.
- Add tests for alias injection, duplicate prevention, and custom alias preservation.

## Verification
- `uloop compile`
- `uloop run-tests --test-mode EditMode --filter-type regex --filter-value "DynamicCodeSourcePreparerTests"`
- `uloop execute-dynamic-code --compile-only true --code "GameObject go = new GameObject(\"compile-only\"); Object.Instantiate(go); return null;"`

<a href="https://cubic.dev/pr/hatayama/unity-cli-loop/pull/1469?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

This PR updates dynamic code preparation/wrapping so Unity-style calls like `Object.Instantiate` work without requiring `UnityEngine.Object`, while preventing ambiguous `Object` references with `System.Object`.

### What changed
- `SourceShaper.Analyze` now centralizes `using`-directive collection via a helper that also extracts alias declarations from `using ... = Alias` (supports optional `global`, inline comments, and verbatim identifier aliases like `using `@Object` = ...`).
- The analysis result now includes `AliasedNames`, which is passed through `DynamicCodeSourcePreparer` into `WrapperTemplate.Build`.
- `WrapperTemplate.Build` now accepts `aliasedNames` and injects default wrapper aliases only when the corresponding alias is requested/needed (currently:
  - `using Object = UnityEngine.Object;`
  - `using Random = UnityEngine.Random;`),
  while using the parsed alias set to avoid duplicate injection when the user already defines `Object` (including verbatim `@Object`) or `Random`.
- `SourceShaper` also refactors comment-aware scanning used during parsing/advancing, improving robustness for alias detection around comments.

### Tests
- Added/extended NUnit coverage for wrapper alias injection and duplicate prevention in `DynamicCodeSourcePreparerTests`:
  - inject `using Object = UnityEngine.Object;` when the snippet uses bare `Object`
  - avoid duplicating the alias when `using Object = UnityEngine.Object;` already exists
  - preserve user-defined aliases like `using Object = System.Object;`
  - respect verbatim aliases like `using `@Object` = System.Object;`
  - inject `using Random = UnityEngine.Random;` when bare `Random` is used
- Added `DynamicCodeTestStringUtility.CountSubstring` and used it to safely assert substring counts (including returning `0` for empty targets).
- Expanded `SourceShaperTests` to verify alias extraction correctness, including `global using` with comments and verbatim alias syntax.
- Updated existing wrapper/pre-using resolver tests to match the new `WrapperTemplate.Build` signature (including added empty alias-collection arguments).

### CODE REVIEW

File: Assets/Tests/Editor/DynamicCodeToolTests/DynamicCodeTestStringUtility.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+namespace io.github.hatayama.uLoopMCP.DynamicCodeToolTests
+{
+    internal static class DynamicCodeTestStringUtility
+    {
+        public static int CountSubstring(string source, string target)
+        {
+            int count = 0;
+            int index = 0;
+            while ((index = source.IndexOf(target, index, System.StringComparison.Ordinal)) >= 0)
```
[the remark this answers is not in the document]
Comment (hatayama) [reply Assets/Tests/Editor/DynamicCodeToolTests/DynamicCodeTestStringUtility.cs:14]: Resolved in 2ce5a96c.

Added an empty-target guard to DynamicCodeTestStringUtility.CountSubstring so the helper returns 0 instead of looping forever when target is empty. Verified with the focused EditMode tests and uloop compile.

File: Assets/Tests/Editor/DynamicCodeToolTests/DynamicCodeSourcePreparerTests.cs [hunk 2]
```diff
@@ -39,6 +39,72 @@ public void Prepare_WhenStringLiteralExists_ShouldEmitPreambleBeforeUserCodeMark
...
+                DynamicCodeConstants.DEFAULT_CLASS_NAME);
+
+            Assert.IsNotNull(prepared.PreparedSource);
+            Assert.AreEqual(
+                1,
+                DynamicCodeTestStringUtility.CountSubstring(prepared.PreparedSource, "using Object = "));
+        }
+
+        [Test]
+        public void Prepare_WhenVerbatimObjectAliasAlreadyExists_ShouldRespectUserAlias()
```
[the remark this answers is not in the document]
Comment (hatayama) [reply Assets/Tests/Editor/DynamicCodeToolTests/DynamicCodeSourcePreparerTests.cs:83]: Resolved in 2ce5a96c.

Added a positive assertion that the user-provided using @Object = System.Object; alias remains in the prepared source, alongside the existing check that the default UnityEngine.Object alias is not injected.
